internal/pkg/manifest/workload.go

```diff
@@ -0,0 +1,291 @@
+			Context:    aws.String(filepath.Join(rootDirectory, filepath.Dir(df))),
+			Args:       i.args(),
+		}
+	}
+	if df == "" && ctx != "" {
+		return &DockerBuildArgs{
+			Dockerfile: aws.String(filepath.Join(rootDirectory, ctx, dockerfileDefaultName)),
+			Context:    aws.String(filepath.Join(rootDirectory, ctx)),
+			Args:       i.args(),
+		}
+	}
+	return &DockerBuildArgs{
+		Dockerfile: aws.String(filepath.Join(rootDirectory, dockerfileDefaultName)),
+		Context:    aws.String(rootDirectory),
+		Args:       i.args(),
+	}
+}
+
+// dockerfile returns the path to the workload's Dockerfile. If no dockerfile is specified,
+// returns "".
+func (i *Image) dockerfile() string {
+	// Prefer to use the "Dockerfile" string in BuildArgs. Otherwise,
+	// "BuildString". If no dockerfile specified, return "".
+	if i.Build.BuildArgs.Dockerfile != nil {
+		return aws.StringValue(i.Build.BuildArgs.Dockerfile)
+	}
+
+	var dfPath string
+	if i.Build.BuildString != nil {
+		dfPath = aws.StringValue(i.Build.BuildString)
+	}
+
+	return dfPath
+}
+
+// context returns the build context directory if it exists, otherwise an empty string.
+func (i *Image) context() string {
+	return aws.StringValue(i.Build.BuildArgs.Context)
+}
+
+// args returns the args section, if it exists, to override args in the dockerfile.
+// Otherwise it returns an empty map.
+func (i *Image) args() map[string]string {
+	return i.Build.BuildArgs.Args
+}
+
+// BuildArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type DockerBuildArgs.
+type BuildArgsOrString struct {
+	BuildString *string
+	BuildArgs   DockerBuildArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the BuildArgsOrString
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (b *BuildArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&b.BuildArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !b.BuildArgs.isEmpty() {
+		// Unmarshaled successfully to b.BuildArgs, return.
+		return nil
+	}
+
+	if err := unmarshal(&b.BuildString); err != nil {
+		return errUnmarshalBuildOpts
+	}
+	return nil
+}
+
+// DockerBuildArgs represents the options specifiable under the "build" field
+// of Docker Compose services. For more information, see:
+// https://docs.docker.com/compose/compose-file/#build
+type DockerBuildArgs struct {
+	Context    *string           `yaml:"context,omitempty"`
+	Dockerfile *string           `yaml:"dockerfile,omitempty"`
+	Args       map[string]string `yaml:"args,omitempty"`
+}
+
+func (b *DockerBuildArgs) isEmpty() bool {
+	if b.Context == nil && b.Dockerfile == nil && b.Args == nil {
+		return true
+	}
+	return false
+}
+
+// LogConfig holds configuration for Firelens to route your logs.
+type LogConfig struct {
```

Comment:
Would you mind renaming this to `Logging` so it's like your `Image` change and the struct name matches the yaml field name.